backend/src/test/service/CachingService.test.ts

```diff
@@ -0,0 +1,41 @@
+import { CachingService } from '../../main/service/CachingService';
+import { CachingObject } from '../../main/data/CachingObject';
+
+describe('simple add/get for redis cache', function () {
+  // NOTE: RUNNING THESE TESTS WILL MODIFY THE REDIS SERVER!
```

Comment:
Hmm yeah this is a difficult one. Essentially, when we run tests, we don't want them to interfere with the cache because the cache may contain sensitive data that we don't want to mix with test data.

This would be a good opportunity to create a mock redis client, so for example, when we call `redis.setex` here, `redis` would actually be a mock instance and `setex` could just be a simple hashmap implementation for the mock class. Note, you'll need to have dependency injection in order to create a mock redis client. Let me know your thoughts, there's a lot of ways we can approach this!

https://github.com/ubclaunchpad/foodies/blob/423ff05514c8b0729ba4e3f4cadad9373d392f58/backend/src/main/service/CachingService.ts#L19-L30

Reply:
Hmm that's a good point! I think one alternative I was also thinking of is just using a local Redis server for our tests? So we would switch in and out which client we're working with

Reply:
Ah yup that would work just as great! Similar to our test database as well.

Reply:
Awesome!! In that case, I'll go with that :)